js/style/style.js

```diff
@@ -633,6 +633,15 @@ Style.prototype = util.inherit(Evented, {
     },
 
     queryRenderedFeatures: function(queryGeometry, params, classes, zoom, bearing) {
+        if (params.filter) {
+            if (validateStyle.emitErrors(this, validateStyle.filter({
+                key: 'queryRenderedFeatures.filter',
+                value: params.filter,
+                style: this.serialize(),
+                styleSpec: styleSpec
+            }))) return this;
```

Comment:
I take that back. There are some places we do throw in `Style`. This should be a throw.